### PR TITLE
ci(hooks): unify commit normalization across local hooks and CI

### PR DESCRIPTION
**Pull request by igor53627**
## Summary

- Add the same 3 normalization transforms (lowercase type, expand aliases, fix spacing) to CI `commit-lint.yml` so PRs from contributors without local hooks still pass
- Replace all unsafe `echo` with `printf '%s\n'` in CI workflow to prevent shell interpretation pitfalls
- Unify exemption syntax to `case` statements matching local `.githooks/prepare-commit-msg` and `.githooks/commit-msg`
- Add `scripts/test_commit_hooks.sh` with 48 table-driven tests covering normalization, validation, exemptions, and end-to-end flows

## Test plan

- [x] Run `scripts/test_commit_hooks.sh` locally (48/48 pass)
- [ ] Verify CI commit-lint job passes on this PR's commits

Nightshift-Task: commit-normalize
Nightshift-Ref: https://github.com/marcus/nightshift

---

> [!NOTE]
> **Low Risk**
> Workflow/doc/test-only changes that adjust commit-message linting behavior; low blast radius but could cause unexpected pass/fail differences for edge-case subjects.
> 
> **Overview**
> CI `commit-lint.yml` now **normalizes commit subjects before validating**, mirroring the local `prepare-commit-msg` hook (lowercase type prefix, expand `feature/bugfix/hotfix/doc` aliases, and insert missing space after the colon) and uses a `case`-based exemption check to match local hooks.
> 
> Adds `scripts/test_commit_hooks.sh` with table-driven tests covering normalization, validation, exemptions, and end-to-end flows, and updates `AGENTS.md` to document the CI normalization behavior and new test script.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b816db881b5248b62c243c71eafd0c184c07ceb3. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **Chores**
  * Improved commit message validation to normalize and validate consistently across local and CI environments.
  * Unified exemption checks so exempted messages behave the same after normalization.
  * Improved validation output and length-failure messaging for clearer CI feedback.
* **Tests**
  * Added a comprehensive table-driven test harness covering normalization, validation, exemptions, and end-to-end flows.
* **Documentation**
  * Updated CI workflow documentation to reflect the normalize-then-validate behavior.

---
*Automated by [nightshift](https://github.com/marcus/nightshift)*

### CODE REVIEW

File: scripts/test_commit_hooks.sh [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    }
+  }')
+
+  # 2. Expand type aliases
+  msg=$(printf '%s\n' "$msg" | sed -E 's/^feature([(:!])/feat\1/')
+  msg=$(printf '%s\n' "$msg" | sed -E 's/^(bugfix|hotfix)([(:!])/fix\2/')
+  msg=$(printf '%s\n' "$msg" | sed -E 's/^doc([(:!])/docs\1/')
+
+  # 3. Fix missing space after colon
+  msg=$(printf '%s\n' "$msg" | sed -E 's/^([a-z]+(\([a-zA-Z0-9_/ -]+\))?!?:)([^ ])/\1 \3/')
```
[the remark this answers is not in the document]
Comment (igor53627) [reply scripts/test_commit_hooks.sh]: Thanks for the suggestion. We are keeping the transforms as separate commands on purpose so the test helper stays line-for-line aligned with local hooks/CI logic (easier drift detection). The functional CI/local mismatch on exemption handling is fixed in 6c5b52d.
